Tests fail to build

While developing a solution to #88 I encountered a problem with the tests. I can't build them.
Here is what I do:
```
CLI11 $ $ git status   
On branch master
Your branch is up-to-date with 'origin/master'.
nothing to commit, working directory clean
CLI11 $ $ git submodule update --init
Submodule 'extern/googletest' (git@github.com:google/googletest.git) registered for path 'extern/googletest'
Submodule 'extern/json' (git@github.com:nlohmann/json.git) registered for path 'extern/json'
Submodule 'extern/sanitizers' (git@github.com:arsenm/sanitizers-cmake) registered for path 'extern/sanitizers'
Submodule path 'extern/googletest': checked out 'ec44c6c1675c25b9827aacd08c02433cccde7780'
Submodule path 'extern/json': checked out 'db53bdac1926d1baebcb459b685dcd2e4608c355'
Submodule path 'extern/sanitizers': checked out '6947cff3a9c9305eb9c16135dd81da3feb4bf87f'
CLI11 $ mkdir build
CLI11 $ cd build
build $ cmake ..
-- The CXX compiler identification is GNU 8.1.0
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- The C compiler identification is GNU 8.1.0
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Found PythonInterp: /usr/bin/python (found version "2.7.12") 
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Looking for pthread_create
-- Looking for pthread_create - not found
-- Looking for pthread_create in pthreads
-- Looking for pthread_create in pthreads - not found
-- Looking for pthread_create in pthread
[- Looking for pthread_create in pthread - found
-- Found Threads: TRUE  
-- Boost version: 1.58.0
-- Configuring done
WARNING: Target "informational" requests linking to directory "/usr/include".  Targets may link only to libraries.  CMake is dropping the item.
-- Generating done
-- Build files have been written to: /home/spirit/src/community/CLI11/build
build $ make
Scanning dependencies of target gtest
[  1%] Building CXX object extern/googletest/googlemock/gtest/CMakeFiles/gtest.dir/src/gtest-all.cc.o
[  2%] Linking CXX static library libgtest.a
[  2%] Built target gtest
...
Scanning dependencies of target OptionalTest
[ 42%] Building CXX object tests/CMakeFiles/OptionalTest.dir/OptionalTest.cpp.o
...
/usr/include/boost/optional/optional.hpp:619:7: error: no matching function for call to 'boost::optional_detail::optional_base<int>::assign_value(std::__cxx11::basic_string<char>&, boost::optional_detail::optional_base<int>::is_reference_predicate)'
       assign_value(boost::forward<Expr>(expr), is_reference_predicate());
...
/usr/include/boost/optional/optional.hpp:608:7: error: cannot convert 'std::__cxx11::basic_string<char>' to 'boost::optional_detail::optional_base<int>::internal_type' {aka 'int'} in initialization
       new (m_storage.address()) internal_type(boost::forward<Expr>(expr)) ;
```

The full log is [attached](https://github.com/CLIUtils/CLI11/files/2879494/CLI11-build.log)

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -8,6 +8,7 @@ Set handling has been completely replaced by a new backend that works as a Valid
 * Support for composable unnamed subcommands [#216]
 * Custom vector separator [#209], [#221]
 * Validators added for IP4 addresses and positive numbers [#210]
+* Minimum required Boost for optional Optionals has been corrected to 1.61
 
 > ### Converting from CLI11 1.7:
 >

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -2,12 +2,12 @@ cmake_minimum_required(VERSION 3.4)
 # Note: this is a header only library. If you have an older CMake than 3.4,
 # just add the CLI11/include directory and that's all you need to do.
 
-# Make sure users don't get warnings on a tested (3.4 to 3.12) version
+# Make sure users don't get warnings on a tested (3.4 to 3.13) version
 # of CMake. For most of the policies, the new version is better (hence the change).
-if(${CMAKE_VERSION} VERSION_LESS 3.12)
+if(${CMAKE_VERSION} VERSION_LESS 3.13)
     cmake_policy(VERSION ${CMAKE_MAJOR_VERSION}.${CMAKE_MINOR_VERSION})
 else()
-    cmake_policy(VERSION 3.12)
+    cmake_policy(VERSION 3.13)
 endif()
 
 set(VERSION_REGEX "#define CLI11_VERSION[ \t]+\"(.+)\"")

--- a/include/CLI/Optional.hpp
+++ b/include/CLI/Optional.hpp
@@ -29,7 +29,7 @@
 
 #if __has_include(<boost/optional.hpp>) && !defined(CLI11_BOOST_OPTIONAL)
 #include <boost/version.hpp>
-#if BOOST_VERSION >= 105800
+#if BOOST_VERSION >= 106100
 #define CLI11_BOOST_OPTIONAL 1
 #endif
 #elif !defined(CLI11_BOOST_OPTIONAL)

--- a/tests/CMakeLists.txt
+++ b/tests/CMakeLists.txt
@@ -110,7 +110,7 @@ file(WRITE "${PROJECT_BINARY_DIR}/CTestCustom.cmake"
     )
 
 # Add boost to test boost::optional if available
-find_package(Boost 1.58)
+find_package(Boost 1.61)
 if(Boost_FOUND)
     if(TARGET Boost::boost)
         target_link_libraries(informational PUBLIC Boost::boost)

--- a/tests/CMakeLists.txt
+++ b/tests/CMakeLists.txt
@@ -112,12 +112,12 @@ file(WRITE "${PROJECT_BINARY_DIR}/CTestCustom.cmake"
 # Add boost to test boost::optional if available
 find_package(Boost 1.58)
 if(Boost_FOUND)
-    if (TARGET Boost::boost)
+    if(TARGET Boost::boost)
         target_link_libraries(informational PUBLIC Boost::boost)
         target_link_libraries(OptionalTest PUBLIC Boost::boost)
     else()
-        target_link_libraries(informational PUBLIC ${Boost_INCLUDE_DIR})
-        target_link_libraries(OptionalTest PUBLIC ${BOOST_INCLUDE_DIR})
+        target_include_directories(informational PUBLIC ${Boost_INCLUDE_DIRS})
+        target_include_directories(OptionalTest PUBLIC ${Boost_INCLUDE_DIRS})
     endif()
     # Enforce Boost::Optional even if __has_include is missing on your compiler
     target_compile_definitions(informational PUBLIC CLI11_BOOST_OPTIONAL)

